main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
Well I really think that the default should respect the current RFC. If you want to be more strict for some non conforming loggers, then it should be optional. I think that ini setting makes most sense in this case.

Also note that you are changing the wrong place as it is not going to work for fpm. Better solution would be to change php_syslog to function and do it in there.